src/main/java/com/example/onlinebookstore/service/shopping/cart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package com.example.onlinebookstore.service.shopping.cart;
+
+import com.example.onlinebookstore.dto.cart.items.CreateCartItemRequestDto;
+import com.example.onlinebookstore.dto.shopping.cart.CartItemUpdateRequestDto;
+import com.example.onlinebookstore.dto.shopping.cart.ShoppingCartDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.ShoppingCartMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.BookRepository;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import com.example.onlinebookstore.repository.UserRepository;
+import jakarta.transaction.Transactional;
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final UserRepository userRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public ShoppingCartDto findById(Long id) {
+        return shoppingCartMapper.toDto(shoppingCartRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException(
+                        "Cannot find shoppingCart by id " + id)
+                )
+        );
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addToCart(Long id, CreateCartItemRequestDto requestDto) {
+        User user = userRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Cannot find user")
+        );
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(id)
+                .orElseGet(() -> {
+                    ShoppingCart shoppingCartNew = new ShoppingCart();
+                    shoppingCartNew.setUser(user);
+                    shoppingCartNew.setCartItems(new HashSet<>());
```

Comment:
you can remove this line if you initialise cartItems by fix comment above in ShoppingCart entity